include/circt/Support/Naming.h

```diff
@@ -0,0 +1,30 @@
+//===- Naming.h - Utilities for handling names ------------------*- C++ -*-===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#ifndef CIRCT_SUPPORT_NAMING_H
+#define CIRCT_SUPPORT_NAMING_H
+
+#include "circt/Support/LLVM.h"
+
+namespace circt {
+
+/// Return true if this is a possibly useless temporary name.
+bool isUselessName(StringRef name);
```

Comment:
one problem here is these heuristics are very firrtl-centric and not obviously correct for general dialects.